ports/unix/main.c

```diff
@@ -562,7 +562,11 @@ MP_NOINLINE int main_(int argc, char **argv) {
             // First entry is empty. We've already added an empty entry to sys.path, so skip it.
             ++path;
         }
-        bool path_remaining = *path;
+        // Allocating path_remaining on the stack can lead to compiler warnings about this
+        // variable being clobbered when nlr is implemented using setjmp/longjmp.  Its
+        // value must be preserved across calls to setjmp/longjmp.
+        static bool path_remaining;
+        path_remaining = *path;
```

Comment:
This fix doesn't look right.  There's not really any setjmp/longjmp involved here.

The original issue actually looks like a compiler bug to me.

Reply:
It was built with size optimisations on, so it might be that the bit of code in question ended in the scope of the setjmp/longjmp further down the code?

I haven't built that with optimisations disabled, I'll check that tomorrow, so I can report the compiler version too.

Reply:
@agatti where you able to check with optimizations disabled, and compiled version?

Reply:
@jonnor nope it flew past my radar.  Anyway, let me give it a try now with the latest Ubuntu Server image - it may have a more recent GCC and make this fix obsolete.

Reply:
@jonnor I tried this on QEMU using the latest Debian Unstable image - it was faster than rebuilding all third party code to build a custom sysroot (my Nezha board is busy doing other things and it's better left alone right now):

With GCC 13.3.0 and GCC 14.1.0, when not applying the patch in question, it fails to build with either optimisations or on off (using `DEBUG=1` in the command line).  On the other hand, I've tried building it with LLVM on the same machine and it builds just fine out of the box.

Considering back in November I used either GCC 12 or 13 (cannot remember the version, it was the default compiler for Ubuntu 23.10 on RISC-V), if it's a compiler bug then I wonder if it has ever been reported (since if it was, it obviously hasn't been addressed in those GCC releases).

@dpgeorge, I'll rewrite the comment block to make things clearer, so this PR will have more chances of being merged and unblock builds on RV64.  GCC is still the default compiler for MicroPython, after all.

Reply:
Thanks for updating the comment, looks good now.